diff -Naur a/data/com.belmoussaoui.Obfuscate.desktop.in.in b/data/com.belmoussaoui.Obfuscate.desktop.in.in
--- a/data/com.belmoussaoui.Obfuscate.desktop.in.in	2023-06-14 10:11:53.662618652 +0530
+++ b/data/com.belmoussaoui.Obfuscate.desktop.in.in	2023-06-14 10:15:37.635616283 +0530
@@ -8,7 +8,7 @@
 # Translators: Search terms to find this application. Do NOT translate or localize the semicolons! The list MUST also end with a semicolon!
 Keywords=Censor;Private;Image;Obfuscate;
 # Translators: Do NOT translate or transliterate this text (this is an icon file name)!
-Icon=@icon@
+Icon=snap.obfuscate.@icon@
 StartupNotify=true
 MimeType=image/png;image/jpeg;image/bmp
 # Translators: Do NOT translate or transliterate this text (these are enum types)!
diff -Naur a/data/com.belmoussaoui.Obfuscate.metainfo.xml.in.in b/data/com.belmoussaoui.Obfuscate.metainfo.xml.in.in
--- a/data/com.belmoussaoui.Obfuscate.metainfo.xml.in.in	2023-06-14 10:12:23.210854914 +0530
+++ b/data/com.belmoussaoui.Obfuscate.metainfo.xml.in.in	2023-06-14 10:16:02.703662328 +0530
@@ -2,6 +2,7 @@
 <!-- Bilal Elmoussaoui 2019 <bilal.elmoussaoui@gnome.org> -->
 <component type="desktop-application">
   <id>@app-id@</id>
+  <icon type="stock">snap.obfuscate.@app-id@</icon>
   <metadata_license>CC0</metadata_license>
   <project_license>GPL-3.0-or-later</project_license>
   <name>Obfuscate</name>
